presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -514,7 +513,7 @@ else if (!result.getBlocked().isDone()) {
         }
         catch (Throwable t) {
             queryStateMachine.transitionToFailed(t);
-            throw Throwables.propagate(t);
+            throw t;
```

Comment:
Assuming no method is cheating here (otherwise a comment should have existed here), change `Throwable t` to `RuntimeException e`).

I was surprised to learn that cheaters exists in our dependency. Raghav recently discussed with me about MySQL library cheats by using `newInstance`.

Reply:
It could also be `Error`, and we want to fail the query for those as well.